### PR TITLE
ci: Test jaeger-v2 as all-in-one in CI

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- Enable testing of jaeger-v2 in all-in-one mode in CI
- Part of #4843

## Description of the changes
- Add `cmd/jaeger-v2/Dockerfile`
- Parameterize `scripts/build-all-in-one-image.sh` to allow running `jaeger-v2` instead of `all-in-one`
- Call this script with `jaeger-v2` from .github/workflows/ci-all-in-one-build.yml

## How was this change tested?
```shell
$ BINARY=jaeger-v2 BRANCH=v2-docker GITHUB_SHA=abcd scripts/build-all-in-one-image.sh pr-only
```

### CODE REVIEW

**Comment by yurishkuro** [comment]
Marked this WIP because it fails to start jaeger-v2 due to what looks like the random ordering of extensions startup sequence, whereas jaegerquery extension can only run after jaegerstorage

File: .github/workflows/ci-all-in-one-build.yml [hunk 1]
```diff
@@ -51,6 +51,10 @@ jobs:
       if: github.ref_name != 'main'
       run: bash scripts/build-all-in-one-image.sh pr-only
 
+    - name: Build and test jaeger-v2 as all-in-one
+      if: github.ref_name != 'main'
+      run: BINARY=jaeger-v2 bash scripts/build-all-in-one-image.sh pr-only
```
Comment (yurishkuro) [inline .github/workflows/ci-all-in-one-build.yml:56]: this only added 39s to the total run, I think it's worth keeping in this workflow for now

File: scripts/build-all-in-one-image.sh [hunk 2]
```diff
@@ -35,35 +43,40 @@ run_integration_test() {
   docker kill "$CID"
 }
 
-if [ "$mode" = "pr-only" ]; then
+if [ -n "$pull_request" ]; then
```
Comment (albertteoh) [inline scripts/build-all-in-one-image.sh:46]: I suggest something like the following to help with readability:

```bash
is_pull_request=true

if [ "$is_pull_request" = true ]; then
if [ "$is_pull_request" = false ]; then
```

File: cmd/jaeger-v2/internal/extension/jaegerquery/server.go [hunk 3]
```diff
@@ -35,6 +38,11 @@ func newServer(config *Config, otel component.TelemetrySettings) *server {
 	}
 }
 
+// Dependencies implements extension.Dependent to ensure this always starts after jaegerstorage extension.
```
Comment (albertteoh) [inline cmd/jaeger-v2/internal/extension/jaegerquery/server.go:41]: 👍🏼

**Review by albertteoh — DISMISSED** [review]

File: go.mod [hunk 4]
```diff
@@ -61,7 +61,7 @@ require (
 	go.opentelemetry.io/collector/exporter/loggingexporter v0.88.0
 	go.opentelemetry.io/collector/exporter/otlpexporter v0.88.0
 	go.opentelemetry.io/collector/exporter/otlphttpexporter v0.88.0
-	go.opentelemetry.io/collector/extension v0.88.0
+	go.opentelemetry.io/collector/extension v0.88.1-0.20231102040825-2e44da36e2c6
```
Comment (yurishkuro) [inline go.mod:64]: pinned to main branch for now, it will get updates once OTEL does a new release, which should happen before our next release in Dec

**Pull request merged**